File: clients/sellingpartner-api-aa-java/src/com/amazon/SellingPartnerAPIAA/AWSAuthenticationCredentialsProvider.java
package com.amazon.SellingPartnerAPIAA;

import lombok.Builder;
import lombok.Data;

/**
 * AWSAuthenticationCredentialsProvider
 */
@Data
@Builder
public class AWSAuthenticationCredentialsProvider {
    /**
     * AWS IAM Role ARN
     */
    private String roleArn;

    /**
     * AWS IAM Role Session Name
     */
    private String roleSessionName;


}

File: clients/sellingpartner-api-aa-csharp/src/Amazon.SellingPartnerAPIAA/AWSAuthenticationCredentialsProvider.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Amazon.SellingPartnerAPIAA
{
    /**
    * AWSAuthenticationCredentialsProvider
    */
    public class AWSAuthenticationCredentialsProvider
    {
        /**
        * AWS IAM Role ARN
        */
        public String RoleArn { get; set; }

        /**
         * AWS IAM Role Session Name
        */
        public String RoleSessionName { get; set; }
    }
}
